src/Easy.cc

```diff
@@ -1612,6 +1612,38 @@ NAN_METHOD(Easy::SetOpt) {
 
         break;
     }
+
+    // check if option is a blob, and the value is correct
+  } else if ((optionId = IsInsideCurlConstantStruct(curlOptionBlob, opt))) {
+#if NODE_LIBCURL_VER_GE(7, 71, 0)
+    if (value->IsNull()) {
+      setOptRetCode = curl_easy_setopt(obj->ch, static_cast<CURLoption>(optionId), NULL);
+    } else if (value->IsString()) {
+      Nan::Utf8String value(value);
+
+      size_t length = static_cast<size_t>(value.length());
+
+      struct curl_blob blob;
+      blob.data = *value;
+      blob.len = length;
+      blob.flags = CURL_BLOB_COPY;
```

Comment:
Do you think this could give a small perf hit down the road (for bigger payloads)? Honest question, I'm ok with just copying the data for now.

Another approach would be to keep the data internally. Check the `ToFree` data structure.

Reply:
I agree it would be ideal to not copy the data if it isn't needed, but I worry more about correctness than performance. Particularly because Buffers are mutable, but also because copying frees us of having to worry about the memory management of all of these containers. Even the CA certs bundle is only around 200 KiB, which isn't too bad. So I think for now, I would prefer to punt on it, although it would probably be worth looking into.